libsolidity/inlineasm/AsmAnalysis.cpp

```diff
@@ -274,6 +274,36 @@ bool AsmAnalyzer::operator()(assembly::FunctionCall const& _funCall)
 	return success;
 }
 
+bool AsmAnalyzer::operator()(Switch const& _switch)
+{
+	bool success = true;
+	set<tuple<LiteralKind, string>> cases;
+
+	for (auto const& _case: _switch.cases)
```

Comment:
Please visit the children in source order. Also, do you visit the case literals?